### PR TITLE
AB#38189 Restrict Export by Group

### PR DESCRIPTION
**Pull request by csmith-he**
[AB#38189](https://hedev.visualstudio.com/76e71f30-b1ad-438a-a096-89ac98712f4e/_workitems/edit/38189)

Adds code to restrict search exports to specified user groups if the ENABLE_EXPORT_FOR_SPECIFIC_GROUPS array is populated.

The restriction is placed in the template and also within the logic to stop anyone from bypassing the restriction.

Dev Peer Reviewer: @khodgkinson-he 
Tester: @Rehaan-Arif 
Merge Lead: @aj-he

### CODE REVIEW

**Review by khodgkinson-he — APPROVED** [review]
Looks good, (loving the python set theory!  ^_^)

**Review by Rehaan-Arif — APPROVED** [review]
This has been tested successfully

File: arches/settings.py [hunk 1]
```diff
@@ -256,10 +256,10 @@
 RESTRICT_CELERY_EXPORT_FOR_ANONYMOUS_USER = False
 
 
-# By setting DISABLE_EXPORT_FOR_ANONYMOUS_USER to True, if the user is attempting
-# to export search results and is not signed in with a user account then the request
-# will not be allowed.
-DISABLE_EXPORT_FOR_ANONYMOUS_USER = False
+# Specify the user groups that are allowed to export results from the application
+# If the array is empty then export will be allowed for ALL users.
```
Comment (aj-he) [inline arches/settings.py]: It is a list, not an array.

Comment (csmith-he) [reply arches/settings.py]: Fixed

**Review by aj-he — APPROVED** [review]
looks good to me

**Pull request merged**